src/Nox.Generator/Application/DomainEventHandlers/DomainEventHandlerGenerator.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.CodeAnalysis;
+using Nox.Generator.Common;
+using Nox.Solution;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Nox.Generator.Application;
+
+internal class DomainEventHandlerGenerator : INoxCodeGenerator
+{
+    public NoxGeneratorKind GeneratorKind => NoxGeneratorKind.Application;
+
+    public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorState codeGeneratorState, GeneratorConfig config)
+    {
+        context.CancellationToken.ThrowIfCancellationRequested();
+
+        if (codeGeneratorState.Solution.Infrastructure?.Messaging is null || codeGeneratorState.Solution.Domain?.Entities is null)
+            return;
+
+        foreach (var (operation, raiseIntegrationEvent, entity) in GroupEntitiesWithDomainEventsByCrudOperation(codeGeneratorState.Solution.Domain.Entities))
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            if (raiseIntegrationEvent)
+            {
+                new TemplateCodeBuilder(context, codeGeneratorState)
+                    .WithClassName($"{entity.Name}{operation}DomainEventHandler")
+                    .WithFileNamePrefix($"Application.DomainEventHandlers")
+                    .WithObject("operation", operation)
+                    .WithObject("entity", entity)
+                    .GenerateSourceCodeFromResource("Application.DomainEventHandlers.DomainEventHandler");
+            }
+        }
+    }
+
+    private IEnumerable<(string Operation, bool RaiseIntegrationEvent, Entity entity)> GroupEntitiesWithDomainEventsByCrudOperation(IEnumerable<Entity> entities)
```

Comment:
```suggestion
    private IEnumerable<(string Operation, bool RaiseIntegrationEvent, Entity entity)> GroupEntitiesWithDomainEventsByOperation(IEnumerable<Entity> entities)
```

Reply:
Missed that one. It's updated now. Thanks.